### DISCUSSION TITLE
behind-the-scenes-of-collection-expressions-part-4-the-spread-element/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#425] [Announcements]
# behind-the-scenes-of-collection-expressions-part-4-the-spread-element/

In this post I show what the compiler generates when you use collection expressions with the spread element and how it depends on the collections used

https://andrewlock.net/behind-the-scenes-of-collection-expressions-part-4-the-spread-element/

**Comment by vyrp** [comment]
Firstly, I'd like to say I'm loving this series! Great content.

I think I spotted a few typos. Going bottom up:
* In the paragraph just before the **Summary** title: should `I'll how how` have been `I'll show how`?
* In the code snippet before the **Summary** title, should it be `readOnlySpan` instead of `span` in the following lines?
    ```
    span.CopyTo(new Span<int>(array).Slice(index, span.Length));
    index += span.Length;
    ```

Finally, I think you can simplify the Linq code from the **Creating IEnumerable<T> using spread element** section using Enumerable.Append:
```
Enumerable.Repeat(0, 1)
    .Concat(source)
    .Append(6)
    .Append(7)
    .Concat(source2)
```

**Comment by KennethHoff** [comment]
I noticed you called `..` the "spread element". This is not entirely true. `..e` is the spread element. https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions

```cs
ReadOnlySpan<int> a = [1,2,3];
ReadOnlySpan<int> b = [..a, 4,5,6];
```

`..a` is the spread element, `..` does not have a name, but everyone (except the C# language team - in official capacity, that is - calls it "spread operator".

**Reply by andrewlock** [reply]
Haha, yeah, you're right, it's _part_ of the spread element. But all MS insist on not calling it the spread operator, e.g. https://devblogs.microsoft.com/dotnet/refactor-your-code-with-collection-expressions/

Though I also agree, in practice, everyone calls it that 😅
